packages/reakit/src/Dialog/__utils/useHideOnClickOutside.ts

```diff
@@ -47,19 +47,26 @@ export function useHideOnClickOutside(
       // triggered the mousedown event. This prevents the dialog from closing
       // by dragging the cursor (for example, selecting some text inside the
       // dialog and releasing the mouse outside of it).
-      if (mouseDownRef.current === event.target && options.hide) {
-        options.hide();
+      if (mouseDownRef.current === event.target) {
+        options.hide?.();
       }
     },
     options.visible && options.hideOnClickOutside
   );
 
+  const document = getDocument(dialogRef.current);
```

Comment:
`dialogRef.current` may not be populated at the time the `useHideOnClickOutside` function is called, so this should be inside the event handler in this case (it could be outside the event handler on the other custom hook because it was still inside `useEffect`.

Reply:
It sounds like getDocument shouldn't accept null as a parameter in that case, should the type be changed? Looking at it's call sites, it currently only accepts null from getActiveElement, and getActiveElement's call sites don't pass in null either.

Reply:
`getDocument` will return the current window's document as a fallback. My understanding is that that's safer as most people will not need the specific `element.ownerDocument` (as far as I know, only those using components inside of an iframe will need it).

But we can discuss it better on #456

Reply:
Ah ok, vscode missed a load of callsites where null unions were being passed in anyway so the change isn't as trivial as I first thought.